src/target/riscv/riscv-011.c

```diff
@@ -1972,9 +1972,14 @@ static int deassert_reset(struct target *target)
 		return wait_for_state(target, TARGET_RUNNING);
 }
 
-static int read_memory(struct target *target, target_addr_t address,
-		uint32_t size, uint32_t count, uint8_t *buffer, uint32_t increment)
+static int read_memory(struct target *target, const riscv_mem_access_args_t args)
 {
+	target_addr_t address = args.address;
+	uint32_t size = args.size;
+	uint32_t count = args.count;
+	uint8_t *buffer = args.buffer.read;
+	uint32_t increment = args.increment;
```

Comment:
Detail for readability: The variables can be made const, if you like.

```suggestion
	const target_addr_t address = args.address;
	const uint32_t size = args.size;
	const uint32_t count = args.count;
	uint8_t *buffer = args.buffer.read;
	const uint32_t increment = args.increment;
```

Reply:
Addressed